src/Compilers/CSharp/Test/Semantic/Semantics/RecordStructTests.cs

```diff
@@ -1292,6 +1399,1821 @@ public unsafe record C
+{
+    bool M()
+    {
+        return X;
+    }
+}
+";
+            var comp = CreateEmptyCompilation(src, parseOptions: TestOptions.RegularPreview, references: new[] { corlibRef });
+            comp.VerifyEmitDiagnostics(
+                // (2,31): warning CS8907: Parameter 'X' is unread. Did you forget to use it to initialize the property with that name?
+                // readonly record struct C(bool X);
+                Diagnostic(ErrorCode.WRN_UnreadRecordParameter, "X").WithArguments("X").WithLocation(2, 31)
+                );
+
+            Assert.Null(comp.GlobalNamespace.GetTypeMember("C").GetMember("X"));
+            var tree = comp.SyntaxTrees.Single();
+            var model = comp.GetSemanticModel(tree, ignoreAccessibility: false);
+            var x = tree.GetRoot().DescendantNodes().OfType<ReturnStatementSyntax>().Single().Expression;
+            Assert.Equal("System.Boolean System.ValueType.X { get; set; }", model.GetSymbolInfo(x!).Symbol.ToTestDisplayString());
+        }
+
+        [Fact]
+        public void StaticCtor()
+        {
+            var src = @"
+System.Console.Write(""static ctor"");
+
+record struct R(int x);
+";
+
+            var comp = CreateCompilation(src);
+            comp.VerifyEmitDiagnostics();
+            CompileAndVerify(comp, expectedOutput: "static ctor");
+        }
+
+        [Fact]
+        public void StaticCtor_ParameterlessPrimaryCtor()
+        {
+            var src = @"
+record struct R()
+{
+    static R() { }
+}
+";
+
+            var comp = CreateCompilation(src);
+            comp.VerifyEmitDiagnostics();
+        }
+
+        [Fact]
+        public void StaticCtor_CopyCtor()
+        {
+            var src = @"
+record struct R()
+{
+    static R(R r) { }
+}
+";
+
+            var comp = CreateCompilation(src);
+            comp.VerifyEmitDiagnostics(
+                // (4,12): error CS0132: 'R.R(R)': a static constructor must be parameterless
+                //     static R(R r) { }
+                Diagnostic(ErrorCode.ERR_StaticConstParam, "R").WithArguments("R.R(R)").WithLocation(4, 12)
+                );
+        }
+
+        [Fact]
+        public void InterfaceImplementation_NotReadonly()
+        {
+            var source = @"
+I r = new R(42);
+r.P2 = 43;
+r.P3 = 44;
+System.Console.Write((r.P1, r.P2, r.P3));
+
+interface I
+{
+    int P1 { get; set; }
+    int P2 { get; set; }
+    int P3 { get; set; }
+}
+record struct R(int P1) : I
+{
+    public int P2 { get; set; } = 0;
+    int I.P3 { get; set; } = 0;
+}
+";
+            var comp = CreateCompilation(source);
+            comp.VerifyEmitDiagnostics();
+            CompileAndVerify(comp, expectedOutput: "(42, 43, 44)", verify: Verification.Skipped /* init-only */);
+        }
+
+        [Fact]
+        public void InterfaceImplementation_Readonly()
+        {
+            var source = @"
+I r = new R(42) { P2 = 43 };
+System.Console.Write((r.P1, r.P2));
+
+interface I
+{
+    int P1 { get; init; }
```

Comment:
>init [](start = 18, length = 4)

Are we testing scenario with 'set' setter and readonly record?

Reply:
See `RecordProperties_01_ReadonlyMismatch`

---
In reply to: [590823438](https://github.com/dotnet/roslyn/pull/51720#discussion_r590823438) [](ancestors = 590823438)

Reply:
> See RecordProperties_01_ReadonlyMismatch

Doesn't look like this is it. See the thread for the previous unit-test. This is a "reversed" scenario.

---
In reply to: [590826341](https://github.com/dotnet/roslyn/pull/51720#discussion_r590826341) [](ancestors = 590826341,590823438)